vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc

```diff
@@ -0,0 +1,339 @@
+
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <utility>
+#include <vector>
+
+#include "base/strings/string_util.h"
+#include "base/strings/string_split.h"
+#include "base/strings/stringprintf.h"
+#include "base/strings/utf_string_conversions.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/media/helper.h"
+#include "bat/ledger/internal/media/github.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace braveledger_media {
+
+GitHub::GitHub(bat_ledger::LedgerImpl* ledger):
+  ledger_(ledger) {
+}
+
+// static
+std::string GitHub::GetUserNameFromUrl(const std::string& path) {
+  if (path.empty()) {
+    return std::string();
```

Comment:
let's use `""` instead of `std::string()` for readability. Please change it in all other places